### PR TITLE
Use Pygments instead of PyColorize to syntax highlight tracebacks

### PR DESCRIPTION
**Pull request by alexmojaki**
To start with I've just stuck in `TerminalFormatter()`. @Carreau How do I insert the correct formatter and style based on user configuration?

### CODE REVIEW

File: IPython/core/ultratb.py [hunk 1]
```diff
@@ -152,23 +150,18 @@ def _format_traceback_lines(lines, Colors, lvals, _line_format):
...
-            line = new_line
-
+        line = stack_line.render(pygmented=True).rstrip('\n') + '\n'
         lineno = stack_line.lineno
         if stack_line.is_current:
             # This is the line with the error
             pad = numbers_width - len(str(lineno))
             num = '%s%s' % (debugger.make_arrow(pad), str(lineno))
-            line = '%s%s%s %s%s' % (Colors.linenoEm, num,
-                                    Colors.line, line, Colors.Normal)
```
Comment (alexmojaki) [inline IPython/core/ultratb.py]: `Colors.line` (typically red) was meant to highlight the current line, but it was completely overridden by PyColorize highlighting every token. With pygments the syntax highlighting overrides `Colors.line` inconsistently, so we remove it. No one should notice a difference unless there's a way to have colors excluding syntax highlighting.

**Comment by Carreau** [comment]
`TBTools` is and instance of `Colorable`, so it should have  a `default_style` that is hooked into the command light. 

To be honest it is a real mess; and `PyColorize.py` will also have some utils; There have been 2 or 3 ways of handling colors through the years; so if in a first step themes are not handled I think it's fine.

**Comment by alexmojaki** [comment]
OK, so I've just looked at the config enough to know when there are no colors. Seems like this should be a mergeable PR now.

**Comment by Carreau** [comment]
Thanks again ! That is awesome !

**Pull request merged**